shared-libs/cht-script-api/src/index.js

```diff
@@ -0,0 +1,61 @@
+
+const ADMIN_ROLE = '_admin';
+const NATIONAL_ADMIN_ROLE = 'national_admin'; // Deprecated: kept for backwards compatibility: #4525
+
+const cache = {
+  userSettingsDoc: null,
+  chtCoreSettingsDoc: null
+};
+
+const setUserSettingsDoc = (userSettingsDoc) => {
+  cache.userSettingsDoc = userSettingsDoc;
+};
+
+const setChtCoreSettingsDoc = (chtCoreSettingsDoc) => {
+  cache.chtCoreSettingsDoc = chtCoreSettingsDoc;
+};
+
+const isAdmin = (user) => {
+  if (!user || !user.roles) {
+    return false;
+  }
+
+  return [ADMIN_ROLE, NATIONAL_ADMIN_ROLE].some(role => user.roles.includes(role));
+};
+
+const hasPermission = (permission) => {
+  if (!cache.chtCoreSettingsDoc
+    || !cache.chtCoreSettingsDoc.permissions
+    || !cache.userSettingsDoc
+    || !cache.userSettingsDoc.roles
+  ) {
+    return false;
+  }
+
+  if (isAdmin(cache.userSettingsDoc)) {
+    // Admin has the permissions automatically.
+    return true;
+  }
+
+  const roles = cache.chtCoreSettingsDoc.permissions[permission];
+
+  if (!roles) {
+    return false;
+  }
+
+  return cache.userSettingsDoc.roles.some(role => roles.includes(role));
+};
+
+const getApi = () => {
+  return {
+    v1: {
+      hasPermission
+    }
+  };
+};
+
+module.exports = {
+  setChtCoreSettingsDoc,
```

Comment:
The `setChtCoreSettingsDoc` and the `setUserSettingsDoc` aren't exposed in the `v1 api`, because these aren't meant for regular users as these affects the final result the `hasPermission`.

Reply:
I've been wanting to remove these functions and cache thing from this script - I'm going to try rewire 🤞

Reply:
I've removed the functions and cache from this library, found a way for tests harness by wrapping this library to pass the user and settings from the test case configuration.
Webapp will handle this library to keep the functions (hasPermissions, hasAnyPermission) synchronous for tasks, targets and contact-summary when user doc and cht setting aren't passed as parameters.